mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py

```diff
@@ -0,0 +1,157 @@
+    deploy_cfg = ctx.cfg
+    dtype = cls_scores[0].dtype
+    device = cls_scores[0].device
+    bbox_decoder = self.bbox_coder.decode
+
+    assert len(cls_scores) == len(bbox_preds)
+    cfg = self.test_cfg if cfg is None else cfg
+    cfg = copy.deepcopy(cfg)
+
+    num_imgs = cls_scores[0].shape[0]
+    featmap_sizes = [cls_score.shape[2:] for cls_score in cls_scores]
+
+    self.mlvl_priors = self.prior_generator.grid_priors(
+        featmap_sizes, dtype=dtype, device=device)
+
+    flatten_priors = torch.cat(self.mlvl_priors)
+
+    mlvl_strides = [
+        flatten_priors.new_full(
+            (featmap_size[0] * featmap_size[1] * self.num_base_priors, ),
+            stride)
+        for featmap_size, stride in zip(featmap_sizes, self.featmap_strides)
+    ]
+    flatten_stride = torch.cat(mlvl_strides)
+
+    # flatten cls_scores, bbox_preds and objectness
+    flatten_cls_scores = [
+        cls_score.permute(0, 2, 3, 1).reshape(num_imgs, -1, self.num_classes)
+        for cls_score in cls_scores
+    ]
+    cls_scores = torch.cat(flatten_cls_scores, dim=1).sigmoid()
+
+    flatten_bbox_preds = [
+        bbox_pred.permute(0, 2, 3, 1).reshape(num_imgs, -1, 4)
+        for bbox_pred in bbox_preds
+    ]
+    flatten_bbox_preds = torch.cat(flatten_bbox_preds, dim=1)
+
+    if objectnesses is not None:
+        flatten_objectness = [
+            objectness.permute(0, 2, 3, 1).reshape(num_imgs, -1)
+            for objectness in objectnesses
+        ]
+        flatten_objectness = torch.cat(flatten_objectness, dim=1).sigmoid()
+        cls_scores = cls_scores * (flatten_objectness.unsqueeze(-1))
+
+    scores = cls_scores
+
+    bboxes = bbox_decoder(flatten_priors[None], flatten_bbox_preds,
+                          flatten_stride)
+
+    post_params = get_post_processing_params(deploy_cfg)
+    max_output_boxes_per_class = post_params.max_output_boxes_per_class
+    iou_threshold = cfg.nms.get('iou_threshold', post_params.iou_threshold)
+    score_threshold = cfg.get('score_thr', post_params.score_threshold)
+    pre_top_k = post_params.get('pre_top_k', -1)
+    keep_top_k = post_params.get('keep_top_k', -1)
+
+    priors = torch.cat(self.mlvl_priors)
+    strides = [
+        priors.new_full((featmap_size.numel() * self.num_base_priors, ),
+                        stride)
+        for featmap_size, stride in zip(featmap_sizes, self.featmap_strides)
+    ]
+    strides = torch.cat(strides)
+    kpt_preds = torch.cat([
+        kpt_pred.permute(0, 2, 3, 1).reshape(
+            num_imgs, -1, self.num_keypoints * 2) for kpt_pred in kpt_preds
+    ],
+                          dim=1)
+    flatten_decoded_kpts = self.decode_pose(priors, kpt_preds, strides)
+
+    vis_preds = torch.cat([
+        vis_pred.permute(0, 2, 3, 1).reshape(num_imgs, -1, self.num_keypoints)
+        for vis_pred in vis_preds
+    ],
+                          dim=1).sigmoid()
+
+    result_list = []
+    pred_bbox = bboxes
```

Comment:
pls. check the naming styles of the variable names, use `pred_bboxes`, `pred_scores`, etc.